adoc/admin-security-ldap.adoc

```diff
@@ -2,7 +2,122 @@
+ROOT_PASSWORD=                                  # Admin Password
+SUFFIX="dc=example,dc=org"                      # Suffix
+DATA_DIR=<PWD>/389_ds_data                      # Directory Server Data
+IMAGE=registry.suse.com/caasp/v4/389-ds:1.4.0   # Container Image 
+----
+
+. Execute `docker` command to deploy 389 Directory Server. This will start non-TLS port (389) and TLS port (636) together with automatically self-signed certificate and key.
++
+----
+docker run -d \
+	-p 389:389 \
+	-p 636:636 \
+	-e INSTANCE_NAME=<INSTANCE_NAME> \
+	-e ROOT_DN=<ROOT_DN> \
+	-e ROOT_PASSWORD=<ROOT_PASSWORD> \
+	-e SUFFIX=<SUFFIX> \
+	-v <DATA_DIR>:/data \
+	--name 389-ds <IMAGE>
+----
+
+=== Deploy 389 Directory Server with external certificate
+
+User can choose to replace their own certificate with following actions:
+
+. Stop running container:
++
+----
+docker stop 389-ds
+----
+
+. Copy external certificate `<rootCA.pem>` to mounted data directory `<DATA_DIR>/config/`
+
+. Bash into container with previous mounted data direcory:
++
+----
+docker run --rm -it \
+  -p 389:389 \
+  -p 636:636 \
+  -v <DATA_DIR>:/data \
+  --entrypoint /bin/bash \
+  --name 389-ds <IMAGE>
+----
+
+. List certificate in `/etc/dirsrv/slapd-<INSTANCE_NAME>`
++
+----
+certutil -L -d /etc/dirsrv/slapd-<INSTANCE_NAME> -f pwdfile.txt
+----
++
+----
+---- SAMPLE ----
+Certificate Nickname                                         Trust Attributes
+                                                             SSL,S/MIME,JAR/XPI
+ 
+Self-Signed-CA                                               CT,,
+Server-Cert                                                  u,u,u
+----
+
+. List key in `/etc/dirsrv/slapd-<INSTANCE_NAME>`
++
+----
+certutil -K -d /etc/dirsrv/slapd-<INSTANCE_NAME> -f pwdfile.txt
+----
++
+----
+---- SAMPLE ----
+certutil: Checking token "NSS Certificate DB" in slot "NSS User Private Key and Certificate Services"
+< 0> rsa      6d4f98efe45328ff852b7e2ca5a24ad46844163a   NSS Certificate DB:Server-Cert
+----
+
+. Delete Existing CA and Certificate in `/etc/dirsrv/slapd-<INSTANCE_NAME>`
++
+----
+certutil -D -d /etc/dirsrv/slapd-<INSTANCE_NAME> -n Self-Signed-CA
+certutil -F -k 6d4f98efe45328ff852b7e2ca5a24ad46844163a -d /etc/dirsrv/slapd-<INSTANCE_NAME> -f pwdfile.txt
+----
+
+. Import certificate in `/etc/dirsrv/slapd-<INSTANCE_NAME>`
++
+----
+openssl pkcs12 -export -inkey <KEY.PEM> -in <CERT.PEM> -out <CERT.PK12> -nodes -name "Server-Cert"
+pk12util -i <CERT.PK12> -d /etc/dirsrv/slapd-<INSTANCE_NAME>
+----
+
+. Import rootCA in `/etc/dirsrv/slapd-<INSTANCE_NAME>`
++
+----
+certutil -A -d /etc/dirsrv/slapd-<INSTANCE_NAME> -n "CA certificate" -t "CT,," -i <rootCA.pem>
+----
+
+. Exit container
+
+. Execute `docker` command to run 389 Directory Server with previous mounted directory:
++
+----
+docker run -d \
+	-p 389:389 \
+	-p 636:636 \
+	-e INSTANCE_NAME=<INSTANCE_NAME> \
+	-e ROOT_DN=<ROOT_DN> \
+	-e ROOT_PASSWORD=<ROOT_PASSWORD> \
+	-e SUFFIX=<SUFFIX> \
```

Comment:
what is the suffix?

Reply:
it is domain suffix, for example "dc=example,dc=com"